### ISSUE TITLE
Override put method in respective classes

### ISSUE THREAD

**Issue by tanweiming00** [nus-cs2030/2122-s2#118]
Question of the day is: 

Did anyone manage not to override put method in their respective classes?

**Comment by wzs02** [comment]
I thought we are supposed to override them? The put method is different in the parent and child classes

**Comment by wjahoward** [comment]
As the put method for each respective class add a different object, the put method has to be different

**Comment by tanweiming00** [comment]
Yeah I override it, was wondering whether anyone managed not to do so. haha

**Comment by ldkai** [comment]
Could this also be an issue of compile time type and run time type? If there is no put method in the child classes at compile time, the methods cannot be invoked

**Comment by szeying02** [comment]
It's technically not considered an override because the return type is different?

**Comment by ReganKvioke** [comment]
I override it as well because...it just seems too much work to do otherwise LOL

**Comment by jlee1007** [comment]
Is it an override because the return type is different? I think we can jus cr8 a new put method
